torch/csrc/autograd/python_variable.cpp

```diff
@@ -565,6 +566,7 @@ static PyObject* THPVariable_make_subclass(PyObject* _ignored, PyObject* args, P
   if (!PyType_Check(cls)) {
     throw torch::TypeError("cls must be a type (got %s)", Py_TYPE(cls)->tp_name);
   }
+  torch_dispatch_mode::StashTorchDispatchModeGuard td_g;
```

Comment:
Should `_make_wrapper_subclass` get the same change?

Reply:
yes, although it does do some disabling rn

Reply:
The issue we were running into is that `detach` was getting redispatched to and making a Python object associated with the tensor. This doesn't happen in wrapper subclass because calling a tensor constructor. I added a test that as written it works with _make_wrapper_subclass in case we change the implementation